src/utils/index.ts

```diff
@@ -0,0 +1,107 @@
+import { SuiClient, SuiEvent } from "@mysten/sui.js/client";
+// import { TransactionResult } from "@mysten/sui.js/transactions";
+
+import { UNIHOUSE_PACKAGE } from "../constants";
+
+interface GenericGameResultInput {
+  coinType: string;
+  moduleName: string;
+  packageId: string;
+  pollInterval?: number;
+  structName: string;
+  suiClient: SuiClient;
+  transactionResult: any;
+}
+
+export interface GenericGameResultResponse {
+    ok: boolean;
+    err?: Error;
+    events?: SuiEvent[];
+}
+
+// Function to sleep for a specified duration
+export const sleep = (ms: number): any => (
+    new Promise(resolve => setTimeout(resolve, ms))
+);
+
+export const extractGenericTypes = (typeName: string): string[] => {
+    const x = typeName.split("<");
+    if (x.length > 1) {
+      return x[1].replace(">", "").replace(" ", "").split(",");
+    } else {
+      return [];
+    };
+};
+
+export const getGenericGameResult = async ({
+  coinType,
+  moduleName,
+  packageId,
+  pollInterval = 3000,
+  suiClient,
+  structName,
+  transactionResult
+}: GenericGameResultInput): Promise<GenericGameResultResponse> => {
+    const result: GenericGameResultResponse = { ok: true };
+
+    try {
+        const objectChanges = transactionResult.objectChanges;
+
+        const gameInfos = (objectChanges as any[])
+            .filter((change) => {
```

Comment:
So only 

`Dice` `Coinflip` RangeDice use the bls_settler event have this event.

Plinko, Limbo, Roulette is actually completetly different change types so a generic function like this won't work

Reply:
Please check a plinko transaction to actually see the string for the event that creates a plinko game.